lib/connection.js

```diff
@@ -669,6 +669,11 @@ Connection.prototype.close = function (callback) {
     }
     setImmediate(callback);
   });
+  // Prevent 'error' listener to be executed before 'close' listener
+  this.netClient.removeAllListeners('error');
+  // Add a noop handler for 'error' event to prevent Socket to throw the error
+  this.netClient.on('error', utils.noop);
```

Comment:
We should subscribe to using `on()` and not `once()` because `'error'` event can be fired after closed, multiple times. ie: if we attempt to call `socket.write()`.